Mathlib/Util/AssertExists.lean

```diff
@@ -52,7 +113,10 @@ You should *not* delete the `assert_not_exists` statement without careful discus
 `assert_not_exists` statements should generally live at the top of the file, after the module doc.
 -/
 elab "assert_not_exists " n:ident : command => do
-  let decl ← try liftCoreM <| realizeGlobalConstNoOverloadWithInfo n catch _ => return
+  let decl ← ((liftCoreM <| realizeGlobalConstNoOverloadWithInfo n) <|> return .anonymous)
+  if decl == .anonymous then
+    Mathlib.AssertNotExist.addDeclEntry true n.getId
```

Comment:
```suggestion
  let decl ← try liftCoreM <| realizeGlobalConstNoOverloadWithInfo n
    catch _ => 
      Mathlib.AssertNotExist.addDeclEntry true n.getId
      return
```

Reply:
:facepalm: Thanks!  I can then remove the awkward `if ... .anonymous ... then ... else`!

Reply:
Actually, `return` does not work, since I should return a name.  I still added the declaration to the environment inside the catch, since that allows to only run the rest of the code in the non-`anonymous` branch.

Reply:
Sorry. I don't understand.

Reply:
Actually, it was I who did not understand: I had added a `:=` before the `←` and then I had to return a `Name`.  Removing the `:=` allowed for the slicker code that you suggested.